### PR TITLE
Adds entitlement and profile checking, fixes other complaints

### PR DESCRIPTION
**Pull request by Kashalls**
Creates checks for fetchEntitlements, fetchProfile that can be passed in from Authflow options. 

Fixes the relyingParty being null when trying to fetch an xbox token without calling a minecraft token function.

Un-exposes endpoints because its not needed.

### CODE REVIEW

**Comment by u9g** [comment]
can you just call this pr what this is and release it in a different pr?

File: src/MicrosoftAuthFlow.js [hunk 1]
```diff
@@ -134,19 +134,41 @@ class MicrosoftAuthFlow {
     }
   }
 
-  async getMinecraftJavaToken () {
+  /**
+   * @typedef {Object} JavaTokenOptions
```
Comment (extremeheat) [inline src/MicrosoftAuthFlow.js:138]: You can put this these in a d.ts file so other programs can actually utilize them

Comment (Kashalls) [reply src/MicrosoftAuthFlow.js:138]: I don't know how to implement this, someone will have to do this for me.

Comment (Kashalls) [reply src/MicrosoftAuthFlow.js:138]: I've implemented the index.d.ts file, just so you know i hated every minute of it.

**Review by extremeheat — APPROVED** [review]

**Pull request merged**